pkg/gce-pd-csi-driver/controller.go

```diff
@@ -328,11 +337,12 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 
 			// Verify that the volume in VolumeContentSource exists.
 			diskFromSourceVolume, err := gceCS.CloudProvider.GetDisk(ctx, project, sourceVolKey, gceAPIVersion)
+			diskTypeForMetric = metrics.GetDiskType(diskFromSourceVolume)
```

Comment:
I don't think we should override the diskType in this case. This is the cloning case and refers to the disk we want to clone from, but not necessarily the target diskType